File: C/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>


int main(void)
{
    string text = get_string("Text: ");
    
    int element = 0;
    
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] >= 'a' && text[i] <= 'z')
        {
            element += 1;
        }
        if (text[i] >= 'A' && text[i] <= 'Z')
        {
            element += 1;
        }
    }
    
    int w = 0; 
    
    if ((isalpha(text[0])))
    {
        w += 1;
    }
    else
    {
        w = 0;
    }
    for (int i = 0; i < strlen(text); i++)
    {
        if ((isspace(text[i])) && (isalpha(text[i + 1])))
        {
            w += 1;
        }
        if (text[i] == '"' && (isalpha(text[i + 1])))
        {
            w += 1;
        }
    }
    
    int sen = 0;
    
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sen += 1;
        }
    }
    
    float L = ((float)element / (float)w) * 100;
    float S = ((float)sen / (float)w) * 100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);
    
    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else 
    {
        printf("Grade %i\n", index);
    }
}

File: Python/readability.py
from cs50 import get_string

text = get_string("Text: ")
letters = 0
for i in range(len(text)):
    if(text[i].isalpha()):
        letters += 1
words = 0
if (text[0].isalpha()):
    words += 1
for i in range(len(text)):
    if(text[i].isspace() and text[i+1].isalpha()):
        words += 1
    if(text[i].isspace() and text[i+1] == '"'):
        words += 1

sentences = 0
for i in range(len(text)):
    if(text[i] == '.' or text[i] == '?' or text[i] == '!'):
        sentences += 1

l = (letters / words) * 100
S = (sentences / words) * 100
index = round(0.0588 * l - 0.296 * S - 15.8)

if index >= 16:
    print("Grade 16+")
elif index < 1:
    print("Before Grade 1")
else:
    print(f"Grade {index}")
